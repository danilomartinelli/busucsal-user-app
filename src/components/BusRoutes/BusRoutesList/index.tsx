import React from "react";

import { FlatList } from "react-native";

import { Route } from "../../../types/Route";
import BusRoutesItem from "../BusRoutesItem";

interface IProps {
  routes: Route[];
  onClickItem: (route: Route) => void;
}

const BusRoutesList = ({ routes, onClickItem }: IProps) => (
  <FlatList
    data={routes}
    renderItem={({ item }) => (
      <BusRoutesItem
        key={item.id}
        onPress={() => onClickItem(item)}
        route={item.route}
        routeHour={item.routeHour}
      />
    )}
  />
);

export default BusRoutesList;
