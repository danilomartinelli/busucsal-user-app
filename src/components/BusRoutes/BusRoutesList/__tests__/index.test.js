import React from "react";
import { shallow } from "enzyme";
import BusRoutesList from "..";

describe("<BusRoutesList />", () => {
  const getProps = () => ({
    routes: [
      {
        id: 1,
        route: "",
        routeHour: new Date().toLocaleDateString(),
        points: [
          {
            lat: 123123,
            lng: 123123,
            name: "dat",
          },
        ],
      },
    ],
    onClickItem: jest.fn(),
  });

  describe("render", () => {
    it("should render properly", () => {
      const propsMock = getProps();

      const wrapper = shallow(<BusRoutesList {...propsMock} />);
      expect(wrapper).toMatchSnapshot();
    });

    test.todo("Test if list renders a item");
  });
});
