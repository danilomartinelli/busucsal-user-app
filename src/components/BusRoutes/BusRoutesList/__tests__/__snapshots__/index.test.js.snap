// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusRoutesList /> render should render properly 1`] = `ShallowWrapper {}`;
